src/java.base/share/classes/jdk/internal/classfile/Classfile.java

```diff
@@ -209,47 +251,44 @@ public static byte[] build(ClassDesc thisClass,
      * @param handler a handler that receives a {@link ClassBuilder}
      * @return the classfile bytes
      */
-    public static byte[] build(ClassEntry thisClassEntry,
-                               ConstantPoolBuilder constantPool,
-                               Consumer<? super ClassBuilder> handler) {
-        thisClassEntry = AbstractPoolEntry.maybeClone(constantPool, thisClassEntry);
-        DirectClassBuilder builder = new DirectClassBuilder((SplitConstantPool)constantPool, thisClassEntry);
-        handler.accept(builder);
-        return builder.build();
-    }
+    byte[] build(ClassEntry thisClassEntry,
+                 ConstantPoolBuilder constantPool,
+                 Consumer<? super ClassBuilder> handler);
 
     /**
      * Build a classfile into a file.
      * @param path the path to the file to write
      * @param thisClass the name of the class to build
      * @param handler a handler that receives a {@link ClassBuilder}
      */
-    public static void buildTo(Path path,
-                               ClassDesc thisClass,
-                               Consumer<ClassBuilder> handler) throws IOException {
-        Files.write(path, build(thisClass, Collections.emptySet(), handler));
+    default void buildTo(Path path,
+                         ClassDesc thisClass,
+                         Consumer<ClassBuilder> handler) throws IOException {
+        Files.write(path, build(thisClass, handler));
     }
 
     /**
-     * Build a classfile into a file.
+     * Build a classfile into a file using the provided constant pool
+     * builder (which encapsulates classfile processing options.)
```

Comment:
The CP is now simply a data holder; it no longer holds the processing options. Same update is needed for the other few build methods taking a CP instance.
```suggestion
     * builder.
```